src/module.c

```diff
@@ -9512,6 +9539,27 @@ void moduleInitModulesSystem(void) {
     pthread_mutex_lock(&moduleGIL);
 }
 
+void modulesCron(void) {
+    /* Check number of temporary clients in the pool and free the unused ones
+     * since the last cron. moduleTempClientMinCount tracks minimum count of
+     * clients in the pool since the last cron. This is the number of clients
+     * that we didn't use for the last cron period. */
+
+    /* Limit the max client count to be freed at once to avoid latency spikes.*/
+    int iteration = 500;
+    /* We are freeing clients if we have more than 16 unused clients. Keeping
+     * small amount of clients to avoid client allocation costs if temporary
+     * clients are required after some idle period. */
+    const int min_client = 16;
+    while (iteration > 0 && moduleTempClientCount > 0 && moduleTempClientMinCount > min_client) {
```

Comment:
the new mechanism of only releasing clients that where not used since the recent cron seems ok to me.
it's a little hard to follow, but i suppose ok.

p.s. why did you change the `iteration` limit from 10 to 500?
original one seemed too low, and this one seems too high.
i did suggest it could be proportional to the number of recently allocated or release clients, but that may be an overkill..
so just asking how you got to that number?

p.p.s. 16, 8, or 4 for `min_client` all seem fine to me..
maybe with a slight preference to the lower ones, so we could settle on 8.

Reply:
Set 8 clients to as min number. 

In previous version, I kept iteration count low to give clients to more chance to be reused. Now, as we track max used count, we can set it higher(we know more or less how many clients we use now). Regarding exact number, freeClient() takes 100-150 nanoseconds on my local on average. So, we need a max limit as freeing 10k clients may cause 1 ms latency spike(maybe more on slower CPUs). I'm okay with any number here, just need an upper limit to prevent worst case scenario (freeing thousands at the same time). Can you suggest one?

Reply:
Sorry, confused by numbers, 500 is a lot. Checking it again, on my local I see freeClient() takes around 800-900 nanoseconds. So,  we need upper limit here.  Changed it to 50, which will introduce 50 microsecond latency on this case. Should we go higher or lower? Can you suggest one?  

ps. Set `min_client` to 8.

Reply:
seems fine :shrug: